### ISSUE TITLE
Updating feature dependencies inside environment breaks PATH

### ISSUE THREAD

**Issue by twrightsman** [prefix-dev/pixi#1360]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

pixi.toml:

```toml
[project]
name = "pixi-test"
version = "0.1.0"
description = "Add a short description here"
authors = ["twrightsman <snip>"]
channels = ["conda-forge"]
platforms = ["linux-64"]

[dependencies]

[feature.feat1.dependencies]
marimo = "~=0.4.11"

[environments]
env1 = { features = ["feat1"] }
```

Reproduction

```
$ pixi shell -e env1
(pixi-test:env1) $ marimo --version
0.4.11
```

Now, bump marimo to `marimo = "~=0.5.0"` in `pixi.toml`.

```
(pixi-test:env1) $ pixi install
✔ Project in <snip>/pixi-test is ready to use!
(pixi-test:env1) $ marimo --version
bash: marimo: command not found
(pixi-test:env1) $ echo $PATH
<snip>/pixi-test/.pixi/envs/default/bin:<snip>
(pixi-test:env1) $ exit
$ pixi shell -e env1
(pixi-test:env1) $ marimo --version
0.5.1
(pixi-test:env1) $ echo $PATH
<snip>/pixi-test/.pixi/envs/env1/bin:<snip>
```

### Issue description

Pixi appears to set the PATH incorrectly during `pixi install` when a feature's dependencies are updated. This does not happen if the dependency is in the default feature. Using `pixi install -e env1` shows the same behavior.

### Expected behavior

The above "command not found" should not occur.

**Comment by ruben-arts** [comment]
Found the issue fixed in #1364

**Comment by twrightsman** [comment]
That's awesome! Thank you for debugging this issue.
